airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry.py

```diff
@@ -148,70 +160,150 @@ def construct_registry_with_spec_from_registry(registry: dict, cached_specs: Out
-    from_metadata_and_spec = construct_registry_with_spec_from_registry(from_metadata, cached_specs)
-    return from_metadata_and_spec
+    registry_name = "oss"
+    registry_directory_manager = context.resources.registry_directory_manager
+
+    from_metadata = construct_registry_from_metadata(metadata_definitions, registry_name)
+    registry_dict = construct_registry_with_spec_from_registry(from_metadata, cached_specs)
+    registry_model = ConnectorRegistryV0.parse_obj(registry_dict)
+
+    file_handle = persist_registry_to_json(registry_model, registry_name, registry_directory_manager)
+
+    metadata = {
+        "gcs_path": MetadataValue.url(file_handle.gcs_path),
+    }
+
+    return Output(metadata=metadata, value=registry_model)
 
 
 @asset(group_name=GROUP_NAME)
-def cloud_sources_dataframe(latest_cloud_registry_dict: dict) -> OutputDataFrame:
-    sources = latest_cloud_registry_dict["sources"]
+def cloud_sources_dataframe(cloud_registry_from_metadata: ConnectorRegistryV0) -> OutputDataFrame:
+    cloud_registry_from_metadata_dict = to_json_sanitized_dict(cloud_registry_from_metadata)
+    sources = cloud_registry_from_metadata_dict["sources"]
     return output_dataframe(pd.DataFrame(sources))
 
 
 @asset(group_name=GROUP_NAME)
-def oss_sources_dataframe(latest_oss_registry_dict: dict) -> OutputDataFrame:
-    sources = latest_oss_registry_dict["sources"]
+def oss_sources_dataframe(oss_registry_from_metadata: ConnectorRegistryV0) -> OutputDataFrame:
+    oss_registry_from_metadata_dict = to_json_sanitized_dict(oss_registry_from_metadata)
+    sources = oss_registry_from_metadata_dict["sources"]
     return output_dataframe(pd.DataFrame(sources))
 
 
 @asset(group_name=GROUP_NAME)
-def cloud_destinations_dataframe(latest_cloud_registry_dict: dict) -> OutputDataFrame:
-    destinations = latest_cloud_registry_dict["destinations"]
+def cloud_destinations_dataframe(cloud_registry_from_metadata: ConnectorRegistryV0) -> OutputDataFrame:
+    cloud_registry_from_metadata_dict = to_json_sanitized_dict(cloud_registry_from_metadata)
+    destinations = cloud_registry_from_metadata_dict["destinations"]
     return output_dataframe(pd.DataFrame(destinations))
 
 
 @asset(group_name=GROUP_NAME)
-def oss_destinations_dataframe(latest_oss_registry_dict: dict) -> OutputDataFrame:
-    destinations = latest_oss_registry_dict["destinations"]
+def oss_destinations_dataframe(oss_registry_from_metadata: ConnectorRegistryV0) -> OutputDataFrame:
+    oss_registry_from_metadata_dict = to_json_sanitized_dict(oss_registry_from_metadata)
+    destinations = oss_registry_from_metadata_dict["destinations"]
     return output_dataframe(pd.DataFrame(destinations))
 
 
-@asset(required_resource_keys={"latest_cloud_registry_gcs_file"}, group_name=GROUP_NAME)
-def latest_cloud_registry_dict(context: OpExecutionContext) -> dict:
-    oss_registry_file = context.resources.latest_cloud_registry_gcs_file
+# Old Registry
```

Comment:
Wdyt about creating a `legacy_registry.py` module to store all the legacy assets. Then we could simply delete it once the migration is over?

Reply:
Love that. Going to do that in my Connector report PR as ive already done something similar ;)